### PR TITLE
Upgrade rules_jvm_external to 5.2

### PR DESCRIPTION
**Pull request by meteorcloudy**
- Upgraded rules_jvm_external version.
- Fixed the patch file to match the latest version.

Fixes https://github.com/bazelbuild/bazel/issues/18596

### CODE REVIEW

**Review by aiuto — APPROVED** [review]
This is really just a band-aid.  Having to patch the internals of rules_jvm_external is unsustainable.

**Comment by meteorcloudy** [comment]
Basically the reason is that during the bootstrap build, we'll need the maven jars accessible via vanilla `filegroup` instead of `jvm_import` which depends on the Java toolchain. rules_jvm_external doesn't provide such a way, therefore we had to patch it to add those `_file` targets.

**Comment by meteorcloudy** [comment]
Encountering a error when building `@rules_jvm_external//private/tools/java/com/github/bazelbuild/rules_jvm_external:rules_jvm_external` during bootstrap
```
Exception in thread "main" java.lang.IllegalArgumentException: error: option -source cannot be used together with --release
``` 

Probably have something to do with the conflict between:
- https://cs.opensource.google/bazel/bazel/+/master:src/java_tools/buildjar/java/com/google/devtools/build/buildjar/javac/JavacOptions.java;l=318
- https://github.com/bazelbuild/rules_jvm_external/blob/643f9c9c8ec7bb5271a3523c07e7d19e93dab238/private/tools/java/com/github/bazelbuild/rules_jvm_external/BUILD#L4-L7

/cc @shs96c

File: third_party/rules_jvm_external.patch [hunk 1]
```diff
@@ -60,68 +60,68 @@ index 901a6b0..d763f01 100644
...
- 
+@@ -1036,7 +1050,7 @@ def _coursier_fetch_impl(repository_ctx):
+     )
+
      repository_ctx.report_progress("Generating BUILD targets..")
 -    (generated_imports, jar_versionless_target_labels) = parser.generate_imports(
 +    (generated_imports, jar_versionless_target_labels, _) = parser.generate_imports(
          repository_ctx = repository_ctx,
-         dep_tree = dep_tree,
+         dependencies = v2_lock_file.get_artifacts(lock_file_contents),
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply third_party/rules_jvm_external.patch:72]: This is irrelevant, I was just update the patch content to match upstream changes, the patch might be a bit confusing.

**Comment by aiuto** [comment]
This needs a green team merge, because 3rd party.

**Comment by iancha1992** [comment]
@meteorcloudy could you please squash the commits to one so we can proceed with the merge? Looks like it's got third party changes.

**Comment by meteorcloudy** [comment]
@iancha1992 Oh, thanks for reminding, I'll split this one into multiple PRs.

**Comment by meteorcloudy** [comment]
@iancha1992 Please merge https://github.com/bazelbuild/bazel/pull/18622 first, thanks!

**Comment by aiuto** [comment]
Thanks!

**Pull request closed without merging**